Go 1.17 - Test failure in gojsonschema time format validation

## Expected Behavior

Test success (works for Go <= 1.16)

## Actual Behavior

Test failure with Go 1.17 rc:

```
--- FAIL: TestFormats (0.07s)
    jsonschema_test.go:102: Test failed : /opa/internal/gojsonschema/testdata/draft7/optional/format/time.json
        validation of time strings.
        only RFC3339 not all of ISO 8601 are valid.
        expects: false, given true
        Schema: {"format":"time"}
        Data: "01:01:01,1111"
```
## Additional Info

A test in [`internal/gojsonschema/testdata/draft7/optional/format/time.json`](https://github.com/open-policy-agent/opa/blob/47feea7cbb25d45b3ecb431fb14289dde4432602/internal/gojsonschema/testdata/draft7/optional/format/time.json) verifies that `01:01:01,1111` **fails** to validate as a time.

Time validation is done here: https://github.com/open-policy-agent/opa/blob/47feea7cbb25d45b3ecb431fb14289dde4432602/internal/gojsonschema/format_checkers.go#L266-L271

However ` _, err := time.Parse("15:04:05", "01:01:01,1111")` returns an error `parsing time "01:01:01,1111": extra text: ",1111"` with Go <= 1.16, and no error with Go 1.17.

This is due to this change in Go: https://github.com/golang/go/commit/f02a26bed0f21de7aeef7a60cf62dac2e3dab737 ("," supported as separator for fractional seconds)

--- a/internal/gojsonschema/testdata/draft7/optional/format/time.json
+++ b/internal/gojsonschema/testdata/draft7/optional/format/time.json
@@ -15,7 +15,7 @@
             },
             {
                 "description": "only RFC3339 not all of ISO 8601 are valid",
-                "data": "01:01:01,1111",
+                "data": "14:30",
                 "valid": false
             }
         ]

